pkg/didcomm/protocol/didexchange/models.go

```diff
@@ -73,10 +73,11 @@ type Invitation struct {
 // Request defines a2a DID exchange request
 // https://github.com/hyperledger/aries-rfcs/tree/master/features/0023-did-exchange#1-exchange-request
 type Request struct {
-	Type       string      `json:"@type,omitempty"`
-	ID         string      `json:"@id,omitempty"`
-	Label      string      `json:"label,omitempty"`
-	Connection *Connection `json:"connection,omitempty"`
+	Type       string            `json:"@type,omitempty"`
+	ID         string            `json:"@id,omitempty"`
+	Label      string            `json:"label,omitempty"`
+	Connection *Connection       `json:"connection,omitempty"`
+	Thread     *decorator.Thread `json:"~thread,omitempty"`
```

Comment:
@rolsonquadras decorators are defined in their own RFCs and can be applied to any message type

Reply:
@llorllale The problem i'm seeing is that the RFC specifically mentions `~thread` decorator in [Exchange Response](https://github.com/hyperledger/aries-rfcs/tree/master/features/0023-did-exchange#attributes-1) and not in [Exchange Request](https://github.com/hyperledger/aries-rfcs/tree/master/features/0023-did-exchange#attributes). So question is - did RFC intentionally omit `~thread` decorator in Exchange Request ? If yes, then we are good with the assumption of adding `~thread` in the model. If no, then it might affect the interop between agents if we add that to our model.

Reply:
I claim that the omission of `~thread` from the Exchange Request in the RFC was accidental. Otherwise, how would I correlate the invitation I sent out with an Exchange Request I'm going to receive later on?

As I type this I realize there *is* one case where the Exchange Request may start the protocol: when it is directed at a public DID.

In any case I opened hyperledger/aries-rfcs#208

Reply:
From the outcome of hyperledger/aries-rfcs#208: turns out we should not expect the `invitation` to be part of any context. Therefore, it is true that we should not expect a `~thread` on the Exchange Request. We need to fix this.

Reply:
@llorllale It means its not mandatory that flow starts from invitation only ? .. it can by any means start with exchange request.. there fore threadID is not required in the exchange request ?

Reply:
"we should not expect the invitation to be part of any context."  Does this means invitation will go out of state handler ?